app/decorators/identity_decorator.rb

```diff
@@ -1,9 +1,32 @@
 IdentityDecorator = Struct.new(:identity) do
+  include ActionView::Helpers::TagHelper
+
   def pretty_event_type
-    I18n.t('event_types.authenticated_at', service_provider: identity.display_name)
+    service_provider_html = if return_to_sp_url.present?
+                              return_to_sp_link
```

Comment:
Is this supposed to be return_to_sp_link rather than return_to_sp_url?

Reply:
It's supposed to be SP link. That returns the service provider name wrapped in a link to the service provider URL.